SystemInformer/proctree.c

```diff
@@ -1451,6 +1486,155 @@ static VOID PhpUpdateProcessNodePriorityBoost(
+                }
+            }
+
+            //PhReleaseQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+            // Destroy the blank string above, before assigneding to this instead
+            PhDereferenceObject(ProcessNode->SvcNames);
+
+            ProcessNode->SvcNames = PhFinalStringBuilderString(&stringBuilder);
+        }
+
+        ProcessNode->ValidMask |= PHPN_SERVICENAMES;
+    }
+}
+
+static VOID PhpUpdateProcessServiceCount(
+    _Inout_ PPH_PROCESS_NODE ProcessNode
+)
+{
+    if (!(ProcessNode->ValidMask & PHPN_SVCCOUNT))		// Bitwise flag for calling this update method
+    {
+        ProcessNode->SvcCount = 0;     // Default to nothing
+
+        PhAcquireQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+        if ((ProcessNode->ProcessItem->ServiceList && ProcessNode->ProcessItem->ServiceList->Count != 0))
+        {
+
+            ULONG count = ProcessNode->ProcessItem->ServiceList->Count;
+            ProcessNode->SvcCount = count;									// Remeber this ULONG count for sorting on it
+            PhPrintInt32(ProcessNode->SvcCountText, count);					// What will be displayed in the column (converting ULONG to WCHAR array (PWSTR))
+        }
+
+        PhReleaseQueuedLockShared(&ProcessNode->ProcessItem->ServiceListLock);
+
+        ProcessNode->ValidMask |= PHPN_SVCCOUNT;
+    }
+}
+
+//
+//   This reads in the newly accessible list (ProcessNode->ProcessItem->ThreadList) from procprv.c that is always storing newly createdthread TIDs
+//
+static VOID PhpUpdateProcessNodeThreadTokenCount(_Inout_ PPH_PROCESS_NODE ProcessNode)
+{
+    if (!(ProcessNode->ValidMask & PHPN_THREADTOKENCOUNT))
+    {
+        ProcessNode->ThreadsWithTokenCounter = 0;
+
+        // I don't think this is neccessary here to check for ThreadListLock, as we are simply reading the list
+        //		it is not being modified
+
+        //PhAcquireQueuedLockShared(&ProcessNode->ProcessItem->ThreadListLock);
+
+        if (ProcessNode->ProcessItem->QueryHandle && ProcessNode->ProcessItem->ThreadList)
+        {
+            for (ULONG i = 0; i < ProcessNode->ProcessItem->ThreadList->Count; i++)
+            {
+                //SYSTEM_THREAD_INFORMATION* thread = ProcessNode->ProcessItem->ThreadList->Items[i];
+
+                // Instead, now just going to store TIDs instead of the entire SYSTEM_THREAD_INFORMATION
+                HANDLE* tid = ProcessNode->ProcessItem->ThreadList->Items[i];           // Basically the thread->ClientId.UniqueThread part of SYSTEM_THREAD_INFORMATION
+
+                //ULONG tidTest = HandleToUlong(thread->ClientId.UniqueThread);
+
+                // See thrdprv.c	:		for (i = 0; i < numberOfThreads; i++)
+
+                // ClientId.UniqueThread is a HANDLE, but its just a Thread ID. However leaving as Handle since PhOpenThread accepts it as one
+                HANDLE threadHandle = NULL;
+                if (NT_SUCCESS(PhOpenThread(&threadHandle, THREAD_QUERY_INFORMATION, tid)))
```

Comment:
This is potentially _a lot_ of work on each update. And per your shortcoming elsewhere (not reaping thread items) you _will_ retrieve thread information that is not associated with a different process eventually.